File: parity-js/src/fixture-driver.js
import { readFileSync } from 'node:fs';
import { Ledger, Reconciler } from './index.js';

const sortKeys = value => Array.isArray(value) ? value.map(sortKeys) : value && typeof value === 'object' ? Object.fromEntries(Object.entries(value).sort(([a], [b]) => a.localeCompare(b)).map(([key, nested]) => [key, sortKeys(nested)])) : value;
const fixture = JSON.parse(readFileSync(process.argv[2], 'utf8'));
const now = Date.parse(fixture.clock);
const ledger = new Ledger({ clock: () => now });
for (const intent of fixture.intents) await ledger.record(intent);
const reconciler = new Reconciler({ ledger, clock: () => now });
const reports = [];
for (const event of fixture.events) { const report = await reconciler.reconcile(event); if (report) reports.push(report); }
console.log(JSON.stringify(sortKeys(reports)));

File: parity-py/parity_py/fixture_driver.py
import json, sys
from pathlib import Path
from parity_py import Ledger, Reconciler, ms

fixture=json.loads(Path(sys.argv[1]).read_text(encoding='utf-8'))
now=ms(fixture['clock'])

async def main():
    ledger=Ledger(clock=lambda: now)
    for intent in fixture['intents']: await ledger.record(intent)
    reconciler=Reconciler(ledger,clock=lambda: now)
    reports=[]
    for event in fixture['events']:
        report=await reconciler.reconcile(event)
        if report: reports.append(report)
    print(json.dumps(reports, sort_keys=True, separators=(',', ':')))

if __name__ == '__main__':
    import asyncio
    asyncio.run(main())
